pkg/controller/secrets/dns.go

```diff
@@ -0,0 +1,105 @@
+package secrets
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/acorn-io/baaah/pkg/router"
+	"github.com/acorn-io/runtime/pkg/config"
+	"github.com/acorn-io/runtime/pkg/labels"
+	"github.com/acorn-io/runtime/pkg/publish"
+	"github.com/acorn-io/runtime/pkg/system"
+	"golang.org/x/exp/slices"
+	corev1 "k8s.io/api/core/v1"
+	netv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func HandleDNSSecret(req router.Request, resp router.Response) error {
+	sec := req.Object.(*corev1.Secret)
+	if sec.Namespace != system.Namespace && sec.Name != system.DNSSecretName {
+		return nil
+	}
+
+	domain := string(sec.Data["domain"])
+	if domain == "" {
+		return fmt.Errorf("acorn dns secret is misconfigured, missing domain")
+	}
+
+	cfg, err := config.Get(req.Ctx, req.Client)
+	if err != nil {
+		return err
+	}
+
+	// The config object includes the acorn domain in the list of ClusterDomains if we are in an enabled state.
+	state := *cfg.AcornDNS
+	if slices.Contains(cfg.ClusterDomains, domain) {
+		state = "enabled"
+	}
+
+	if strings.EqualFold(state, "disabled") {
+		return nil
+	}
+
+	ingressClassName := cfg.IngressClassName
+	if ingressClassName == nil || *ingressClassName == "" {
```

Comment:
Is this always non-nil when getting the config?

Reply:
Yes, it is. You are correct and I'll make a change to make the logic less verbose.